src/App.tsx

```diff
@@ -1,5 +1,16 @@
+import { Route, Routes } from 'react-router-dom';
+import HomePage from './Pages/HomePage/HomePage';
+import AuthenticationPage from './Pages/AuthenticationPage/AuthenticationPage';
+
 function App() {
-  return <div className="App">Hello worldegigit branch lll</div>;
+  return (
+    <div className="App">
+      <Routes>
+        <Route element={<HomePage />} path="/" />
+        <Route element={<AuthenticationPage />} path="/Authentication" />
```

Comment:
the authentication process:
1 people get redirected from the url in the anchor tag in the home page to a specific url that reddit specifies
2 after granting access people get redirected to our app with query attached to the request and we should parse it to get the code
3 after retieving the code in the query will send a post request to the www.redit.com/access_token 

The reason why I created a seperate file for the authentication is not to call the useffect every time the main page loads because we need to run the process only when a user is redirected by reddit

I can moove the logic to main if you found it the best to do after reading this as I created a custom hook to handle the logic